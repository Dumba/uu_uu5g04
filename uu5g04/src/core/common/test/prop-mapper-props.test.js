/**
 * Copyright (C) 2019 Unicorn a.s.
 * 
 * This program is free software; you can use it under the terms of the UAF Open License v01 or
 * any later version. The text of the license is available in the file LICENSE or at www.unicorn.com.
 * 
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
 * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See LICENSE for more details.
 * 
 * You may contact Unicorn a.s. at address: V Kapslovne 2767/2, Praha 3, Czech Republic or
 * at the email: info@unicorn.com.
 */

import React from 'react';
import {shallow, mount} from 'enzyme';
import UU5 from "uu5g04";
import "uu5g04-bricks";
import TestTools from "../../test/test-tools.js";

const mockData = {
  name: 'John',
  surname: 'Doe',
  age: 25,
  country: 'USA',
  state: 'California',
  city: 'Los Angeles'
}

const TagName = "UU5.Common.PropMapper";

const CONFIG = {
  mixins: [
    "UU5.Common.BaseMixin"
  ],
  props: {
    mapping: {
      values: [{ "surname": "header", "name": "content", "state": "footer" }, { "city": "header", "state": "content", "country": "footer" }]
    }
  },
  requiredProps: {
    data: mockData,
    children: <UU5.Bricks.Container />
  },
  opt: {
    shallowOpt: {
      disableLifecycleMethods: false
    },
    enzymeToJson: false
  }
};

describe(`${TagName} props testing`, () => {
  TestTools.testProperties(TagName, CONFIG);
});

describe(`${TagName} custom props testing`, () => {
  test("with loader", () => {
    const onLoadFn = dtoIn => {
      dtoIn.done(mockData)
    };

    const wrapper = mount(
      <UU5.Common.Loader onLoad={onLoadFn}>
        <UU5.Common.PropMapper mapping={{ "name": "content" }}>
          <UU5.Bricks.Row />
        </UU5.Common.PropMapper>
      </UU5.Common.Loader>
    );
    wrapper.update();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
