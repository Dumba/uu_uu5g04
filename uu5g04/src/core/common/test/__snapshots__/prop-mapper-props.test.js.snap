// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UU5.Common.PropMapper custom props testing with loader 1`] = `"<div class=\\"uu5-bricks-row uu5-bricks-row-standard uu5-bricks-row-spacing\\">John</div>"`;

exports[`UU5.Common.PropMapper props testing className 1`] = `
<Container
  age={25}
  city="Los Angeles"
  className={null}
  content={null}
  controlled={true}
  country="USA"
  disabled={false}
  dynamic={false}
  footer={null}
  header={null}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name="John"
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  state="California"
  style={null}
  surname="Doe"
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing default props 1`] = `
<Container
  age={25}
  city="Los Angeles"
  className={null}
  content={null}
  controlled={true}
  country="USA"
  disabled={false}
  dynamic={false}
  footer={null}
  header={null}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name="John"
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  state="California"
  style={null}
  surname="Doe"
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing id 1`] = `
<Container
  age={25}
  city="Los Angeles"
  className={null}
  content={null}
  controlled={true}
  country="USA"
  disabled={false}
  dynamic={false}
  footer={null}
  header={null}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name="John"
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  state="California"
  style={null}
  surname="Doe"
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing mainAttrs 1`] = `
<Container
  age={25}
  city="Los Angeles"
  className={null}
  content={null}
  controlled={true}
  country="USA"
  disabled={false}
  dynamic={false}
  footer={null}
  header={null}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name="John"
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  state="California"
  style={null}
  surname="Doe"
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing mapping 1`] = `
<Container
  className={null}
  content="John"
  controlled={true}
  disabled={false}
  dynamic={false}
  footer="California"
  header="Doe"
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name={null}
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  style={null}
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing mapping 2`] = `
<Container
  className={null}
  content="California"
  controlled={true}
  disabled={false}
  dynamic={false}
  footer="USA"
  header="Los Angeles"
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name={null}
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  style={null}
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing mapping 3`] = `
<Container
  className={null}
  content="California"
  controlled={true}
  disabled={false}
  dynamic={false}
  footer="USA"
  header="Los Angeles"
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name={null}
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  style={null}
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing mapping 4`] = `
<Container
  className={null}
  content="John"
  controlled={true}
  disabled={false}
  dynamic={false}
  footer="California"
  header="Doe"
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name={null}
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  style={null}
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing name 1`] = `
<Container
  age={25}
  city="Los Angeles"
  className={null}
  content={null}
  controlled={true}
  country="USA"
  disabled={false}
  dynamic={false}
  footer={null}
  header={null}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name="John"
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  state="California"
  style={null}
  surname="Doe"
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing noIndex 1`] = `
<Container
  age={25}
  city="Los Angeles"
  className={null}
  content={null}
  controlled={true}
  country="USA"
  disabled={false}
  dynamic={false}
  footer={null}
  header={null}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name="John"
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  state="California"
  style={null}
  surname="Doe"
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing noIndex 2`] = `
<Container
  age={25}
  city="Los Angeles"
  className={null}
  content={null}
  controlled={true}
  country="USA"
  disabled={false}
  dynamic={false}
  footer={null}
  header={null}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name="John"
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  state="California"
  style={null}
  surname="Doe"
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing noIndex 3`] = `
<Container
  age={25}
  city="Los Angeles"
  className={null}
  content={null}
  controlled={true}
  country="USA"
  disabled={false}
  dynamic={false}
  footer={null}
  header={null}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name="John"
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  state="California"
  style={null}
  surname="Doe"
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing noIndex 4`] = `
<Container
  age={25}
  city="Los Angeles"
  className={null}
  content={null}
  controlled={true}
  country="USA"
  disabled={false}
  dynamic={false}
  footer={null}
  header={null}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name="John"
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  state="California"
  style={null}
  surname="Doe"
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing style 1`] = `
<Container
  age={25}
  city="Los Angeles"
  className={null}
  content={null}
  controlled={true}
  country="USA"
  disabled={false}
  dynamic={false}
  footer={null}
  header={null}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name="John"
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  state="California"
  style={null}
  surname="Doe"
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing style 2`] = `
<Container
  age={25}
  city="Los Angeles"
  className={null}
  content={null}
  controlled={true}
  country="USA"
  disabled={false}
  dynamic={false}
  footer={null}
  header={null}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name="John"
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  state="California"
  style={null}
  surname="Doe"
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing style 3`] = `
<Container
  age={25}
  city="Los Angeles"
  className={null}
  content={null}
  controlled={true}
  country="USA"
  disabled={false}
  dynamic={false}
  footer={null}
  header={null}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name="John"
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  state="California"
  style={null}
  surname="Doe"
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing style 4`] = `
<Container
  age={25}
  city="Los Angeles"
  className={null}
  content={null}
  controlled={true}
  country="USA"
  disabled={false}
  dynamic={false}
  footer={null}
  header={null}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name="John"
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  state="California"
  style={null}
  surname="Doe"
  tooltip={null}
  underline={false}
/>
`;

exports[`UU5.Common.PropMapper props testing tooltip 1`] = `
<Container
  age={25}
  city="Los Angeles"
  className={null}
  content={null}
  controlled={true}
  country="USA"
  disabled={false}
  dynamic={false}
  footer={null}
  header={null}
  hidden={false}
  id={null}
  ignoreInnerHTML={false}
  key=".0"
  level={null}
  mainAttrs={null}
  name="John"
  nestingLevel={null}
  noIndex={false}
  noSpacing={false}
  pureRender={false}
  ref_={null}
  selected={false}
  state="California"
  style={null}
  surname="Doe"
  tooltip={null}
  underline={false}
/>
`;
